src/Workspaces/Core/Portable/ReassignedVariable/AbstractReassignedVariableService.cs

```diff
@@ -162,22 +186,54 @@ bool ComputeParameterIsAssigned(IParameterSymbol parameter)
                 if (methodOrProperty.DeclaringSyntaxReferences.Length == 0)
                     return false;
 
-                // Be resilient to cases where the parameter might have multiple locations.  This
-                // should not normally happen, but we want to be resilient in case it occurs in
-                // error scenarios.
-                var methodOrPropertyDeclaration = methodOrProperty.DeclaringSyntaxReferences.First().GetSyntax(cancellationToken);
-                if (methodOrPropertyDeclaration.SyntaxTree != semanticModel.SyntaxTree)
-                    return false;
+                if (parameter.IsPrimaryConstructor(cancellationToken))
+                {
+                    // A primary constructor parameter is analyzed across all parts of the type declaration it is
+                    // created in.
```

Comment:
because of this, we may have to grab other semantic models.  that's why the semantic model is passed along to everything.